src/Core/Domain/Category/Command/AbstractAddCategoryCommand.php

```diff
@@ -0,0 +1,295 @@
+    private $localizedMetaDescriptions;
+
+    /**
+     * @var string[]|null
+     */
+    private $localizedMetaKeywords;
+
+    /**
+     * @var int[]|null
+     */
+    private $associatedGroupIds;
+
+    /**
+     * @var int[]|null
+     */
+    private $associatedShopIds;
+
+    /**
+     * @param string[] $name
+     * @param string[] $linkRewrite
+     * @param bool $isActive
+     *
+     * @throws CategoryConstraintException
+     */
+    public function __construct(array $name, array $linkRewrite, bool $isActive)
+    {
+        $this
+            ->setLocalizedNames($name)
+            ->setLocalizedLinkRewrites($linkRewrite)
+            ->setIsActive($isActive);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getLocalizedNames(): array
+    {
+        return $this->localizedNames;
+    }
+
+    /**
+     * @param string[] $localizedNames
+     *
+     * @return $this
+     *
+     * @throws CategoryConstraintException
+     */
+    public function setLocalizedNames(array $localizedNames): self
+    {
+        if (empty($localizedNames)) {
+            throw new CategoryConstraintException('Category name cannot be empty', CategoryConstraintException::EMPTY_NAME);
+        }
+
+        $this->localizedNames = $localizedNames;
+
+        return $this;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getLocalizedLinkRewrites(): array
+    {
+        return $this->localizedLinkRewrites;
+    }
+
+    /**
+     * @param string[] $localizedLinkRewrites
+     *
+     * @return $this
+     *
+     * @throws CategoryConstraintException
+     */
+    public function setLocalizedLinkRewrites(array $localizedLinkRewrites): self
+    {
+        if (empty($localizedLinkRewrites)) {
+            throw new CategoryConstraintException('Category link rewrite cannot be empty', CategoryConstraintException::EMPTY_LINK_REWRITE);
+        }
+
+        $this->localizedLinkRewrites = $localizedLinkRewrites;
+
+        return $this;
+    }
+
+    /**
+     * @return string[]|null
+     */
+    public function getLocalizedDescriptions(): ?array
+    {
+        return $this->localizedDescriptions;
+    }
+
+    /**
+     * @param string[] $localizedDescriptions
+     *
+     * @return $this
+     */
+    public function setLocalizedDescriptions(array $localizedDescriptions): self
+    {
+        $this->localizedDescriptions = $localizedDescriptions;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+        return $this->isActive;
+    }
+
+    /**
+     * @param bool $isActive
+     *
+     * @return $this
+     *
+     * @throws CategoryConstraintException
+     */
+    public function setIsActive(bool $isActive): self
+    {
+        if (!is_bool($isActive)) {
+            throw new CategoryConstraintException('Invalid Category status supplied', CategoryConstraintException::INVALID_STATUS);
+        }
+
+        $this->isActive = $isActive;
```

Comment:
```suggestion
        $this->isActive = $isActive;
```
It's useless and can't happen